src/tike/communicators/pool.py

```diff
@@ -124,4 +124,4 @@ def f(worker, *args):
             with cp.cuda.Device(worker):
                 return func(*args, **kwargs)
 
-        return super().map(f, self.workers, *iterables)
+        return list(super().map(f, self.workers, *iterables))
```

Comment:
Note that I added this call to `list()` because `ThreadPool.map()` actually returns a generator, but our API always assumes a list. i.e. we try to index the result of `map()` directly instead of always iterating over it. The only consequence of calling `list()` here is that it forces a synchronization between the threads, but I'm not sure that for our use case, this matters.

Reply:
It won't be an issue. We anyway need to do a device-device synchronization after each communication; moving it to thread-level is OK.